src/reducers/filter.js

```diff
@@ -0,0 +1,15 @@
+import { CHANGE_FILTER } from '../actions/index';
+
+const filterReducer = (state = { filter: 'All', filteredBooks: [] }, action) => {
+  switch (action.type) {
+    case CHANGE_FILTER:
+      return {
+        filter: action.filter,
+        filteredBooks: action.filter === 'All' ? action.books : action.books.filter((el) => el.category === action.filter),
```

Comment:
- [ ] There is no need to save the actually filtered books here. For simplicity, we can save only the filter. On the BookList component, we can use the filter value to list only the books that belong to the selected filter.